frontend/src/metabase/parameters/utils/dashboards.ts

```diff
@@ -0,0 +1,273 @@
+  return parameterExistsOnDashboard && !parameterHasMapping;
+}
+
+export function getDashboardUiParameters(
+  dashboard: Dashboard,
+  metadata: Metadata,
+): UiParameter[] {
+  const { parameters, ordered_cards } = dashboard;
+  const mappings = getMappings(ordered_cards);
+  const uiParameters: UiParameter[] = parameters.map(parameter => {
+    if (isFieldFilterParameter(parameter)) {
+      return buildFieldFilterUiParameter(parameter, mappings, metadata);
+    }
+
+    return {
+      ...parameter,
+    };
+  });
+
+  return uiParameters;
+}
+
+function buildFieldFilterUiParameter(
+  parameter: Parameter,
+  mappings: Mapping[],
+  metadata: Metadata,
+): FieldFilterUiParameter {
+  const mappingsForParameter = mappings.filter(
+    mapping => mapping.parameter_id === parameter.id,
+  );
+  const mappedFields = mappingsForParameter.map(mapping => {
+    const { target, card } = mapping;
+    return getTargetField(target, card, metadata);
+  });
+  const fields = mappedFields.filter((field): field is Field => field != null);
+  const hasOnlyFieldTargets =
+    mappedFields.length !== 0 && mappedFields.length === fields.length;
+  const uniqueFieldsWithFKResolved = _.uniq(
+    fields.map(field => field.target ?? field),
+    field => field.id,
+  );
+
+  return {
+    ...parameter,
+    fields: uniqueFieldsWithFKResolved,
+    hasOnlyFieldTargets,
+  };
+}
+
+function getMappings(ordered_cards: DashboardOrderedCard[]): Mapping[] {
+  return ordered_cards.flatMap(dashcard => {
+    const { parameter_mappings = [], card, series } = dashcard;
+    const cards = [card, ...(series || [])];
+    return parameter_mappings
+      .map(parameter_mapping => {
+        const card = _.findWhere(cards, { id: parameter_mapping.card_id });
+        return card
+          ? {
+              ...parameter_mapping,
+              dashcard_id: dashcard.id,
+              card,
+            }
+          : null;
+      })
+      .filter((mapping): mapping is Mapping => mapping != null);
+  });
+}
+
+export function getTargetField(
+  target: ParameterVariableTarget | ParameterDimensionTarget,
+  card: SavedCard,
+  metadata: Metadata,
+) {
+  if (!card.dataset_query) {
+    return null;
+  }
+
+  const question = new Question(card, metadata);
+  const field = getParameterTargetField(target, metadata, question);
+  return field ?? null;
+}
+
+export function getParametersMappedToDashcard(
+  dashboard: Dashboard,
+  dashcard: DashboardOrderedCard,
+): ParameterWithTarget[] {
+  const { parameters } = dashboard;
+  const { parameter_mappings = [] } = dashcard;
+  return parameters
+    .map(parameter => {
+      const mapping = _.findWhere(parameter_mappings, {
+        parameter_id: parameter.id,
+      });
+
+      if (mapping) {
+        return {
+          ...parameter,
+          target: mapping.target,
+        };
+      }
+    })
+    .filter((parameter): parameter is ParameterWithTarget => parameter != null);
+}
+
+export function hasMatchingParameters({
```

Comment:
I had to tweak this function because it relied on a function that I deleted (the old `getMappingsByParameterId`). It is well unit tested.